plugins/out_stackdriver/stackdriver.c

```diff
@@ -288,6 +288,30 @@ static bool validate_msgpack_unpacked_data(msgpack_object root)
            root.via.array.ptr[1].type == MSGPACK_OBJECT_MAP;
 }
 
+void replace_prefix_dot(flb_sds_t s, int tag_prefix_len)
+{
+    int str_len;
+    char c;
+
+    if (!s) {
+        return;
+    }
+
+    str_len = flb_sds_len(s);
+    if (tag_prefix_len > str_len) {
+        flb_error("[output] tag_prefix shouldn't be longer than local_resource_id");
+        return;
+    }
+
+    for (int i = 0; i < tag_prefix_len; i++) {
```

Comment:
defining a variable inside the loop will break old compilers. Please move the variable declaration at top

Reply:
updated